app/scripts/modules/ecs/src/serverGroup/configure/wizard/horizontalScaling/horizontalScaling.component.html

```diff
@@ -1,5 +1,86 @@
 <div class="container-fluid form-horizontal">
   <div class="form-group">
+    <div class="col-md-4 sm-label-right">
+      <b>Compute options</b>
+      <help-field key="ecs.computeOptions"></help-field>
+    </div>
+    <div class="col-md-2 radio">
+      <label>
+        <input
+          data-test-id="ServerGroup.computeOptionsLaunchType"
+          type="radio"
+          ng-model="$ctrl.capacityProviderState.useCapacityProviders"
+          ng-value="false"
+          ng-click="$ctrl.capacityProviderState.updateComputeOption('launchType')"
+          id="computeOptionsLaunchType"
+        />
+        Launch type
+      </label>
+    </div>
+    <div class="col-md-1 radio">
+      <label>
+        <input
+          data-test-id="ServerGroup.computeOptionsCapacityProviders"
+          type="radio"
+          ng-model="$ctrl.capacityProviderState.useCapacityProviders"
+          ng-value="true"
+          ng-click="$ctrl.capacityProviderState.updateComputeOption('capacityProviders')"
+          id="computeOptionsCapacityProviders"
+        />
+        Capacity Providers
+      </label>
+    </div>
+  </div>
+
+  <div class="form-group" ng-if="$ctrl.capacityProviderState.useCapacityProviders">
+    <div class="sm-label-left">
+      <b>Capacity Provider Strategies</b>
+      <help-field key="ecs.capacityProviderStrategies"></help-field>
+    </div>
+    <div>
+      <div>{{ cp }}</div>
+      <table class="table table-condensed packed tags">
+        <thead>
+          <th style="width: 50%">Provider name <help-field key="ecs.capacityProviderName"></help-field></th>
+          <th style="width: 25%">Base <help-field key="ecs.capacityProviderBase"></help-field></th>
+          <th style="width: 25%">Weight <help-field key="ecs.capacityProviderWeight"></help-field></th>
+        </thead>
+        <tbody>
+          <tr ng-repeat="cp in $ctrl.command.capacityProviderStrategies">
```

Comment:
I guess we need to have an array of strings here as we are just getting CP names and not the strategies. Also, we need to have a separate parameter for default capacity provider strategy as well. 
Default CP strategy will be coming in the form of 
```
"defaultCapacityProviderStrategy": [ 
            { 
               "base": number,
               "capacityProvider": "string",
               "weight": number
            }
         ]
```

And CPs will be coming in the form of 
 `"capacityProviders": [ "string" ]`

Does that make sense?

Reply:
let's be careful not to conflate the data we need to _submit to clouddriver_ in order to use capacity providers (the "command" obj) and the data we need to _configure the UI_ for capacity providers. 

right now this logic is only displaying what's present in the stage JSON and providing basic text inputs for users to directly update it (i.e., what should be present in the stage JSON). That data should match what we need to submit to clouddriver and will always look like this:

```
"capacityProviderStrategies": [
        {
          "base": "1",
          "capacityProvider": "FARGATE_SPOT",
          "weight": "1"
        }
]
```

You're right that in the near future, we want to enhance the UI and provide a drop-down for `capacityProviderStrategy.name` so users can select CPs associated with their cluster. 

The API call to retrieve that data per cluster returns a more complex object which includes info like `defaultCapacityProviderStrategy`, as you said. But that data will be loaded by the controller and formatted into a list of strings which will populate the drop-down, it will not be added to the stage JSON directly. The drop-down will only update the `capacityProviderStrategy.name` in the stage JSON with the selected value when a user interacts with it and selects something. Let me know if that doesn't clarify things!